### PR TITLE
deploy an app via arguments

### PR DESCRIPTION
**Pull request by jamesgorrie**
I've moved things to [modules](https://www.terraform.io/docs/modules/usage.html) so that we might abstract out into multiple deploys. 

We'll see how far this get's once I have another project to deploy here i.e. the website and patterns.

Seems to work so far.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
@wellcometrust/web-platform

**Comment by jamesgorrie** [comment]
@alfraser86

File: terraform/templates/elb.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  listener {
+    instance_port       = 8080
+    instance_protocol   = "http"
+    lb_port             = 8080
+    lb_protocol         = "http"
+  }
+
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 10
```
Comment (alfraser86) [inline terraform/templates/elb.tf]: I think I had this as 10 when I did it but now wondering if that's too high?

File: terraform/wellcome-patterns/wellcome-collections.tf [hunk 2]
```diff
@@ -0,0 +1,23 @@
+variable "aws_access_key" {}
+variable "aws_secret_key" {}
+variable "web_platform_key_path" {}
+variable "web_platform_key_name" {}
+variable "build_number" {}
+variable "build_bucket" {}
+variable "build_branch" {}
+
+module "wellcome-patterns" {
```
Comment (alfraser86) [inline terraform/wellcome-patterns/wellcome-collections.tf]: Wonder if this name should be more specific to the collection

File: terraform/wellcome-patterns/wellcome-collections.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  source                 = "../templates/"
+  project_name           = "wellcome-patterns"
+  launch_template        = "${file("launch-script.tpl")}"
+  aws_region             = "eu-west-1"
+  aws_availability_zones = "eu-west-1a"
+  aws_node_ami           = "ami-ffaad08c"
```
Comment (alfraser86) [inline terraform/wellcome-patterns/wellcome-collections.tf]: Have you tried setting the ami through the vars file? Then I guess you could have packer switching the ami with bash

File: terraform/templates/sg.tf [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  name        = "ssh-in"
+  description = "Used to allow SSH access"
+  vpc_id      = "${aws_vpc.web_platform.id}"
+
+  # HTTP SSH from anywhere
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (jamesgorrie) [inline terraform/templates/sg.tf]: You need to have [the key](https://github.com/wellcometrust/web-platform-infra/pull/2/files#diff-eaec25a5502b158debd0112c6fd710ceR12) to get in, so I don't see a problem with it - what if I was working remotely? do we have VPN access?

File: terraform/templates/elb.tf (hunk 1 above)
Comment (jamesgorrie) [inline terraform/templates/elb.tf]: This is way to high, it was for testing - what do you think are sensible values?

What about:
```
health_check {
  healthy_threshold   = 2
  unhealthy_threshold = 5
  timeout             = 20
  interval             = 30
}
```

File: terraform/wellcome-patterns/wellcome-collections.tf (hunk 2 above)
Comment (jamesgorrie) [inline terraform/wellcome-patterns/wellcome-collections.tf]: Given that's what we're working on - probably.
I've got a meeting with the team where we can chat about that this afternoon.

File: terraform/templates/sg.tf (hunk 4 above)
Comment (alfraser86) [inline terraform/templates/sg.tf]: We don't but I think we probably should. Then we could lock down ssh to the vpn and Wellcome IPs.

**Pull request merged**